### ISSUE TITLE
Vale should tell me about missing styles

### ISSUE THREAD

**Issue by curiousleo** [vale-cli/vale#159]
- OS: Linux (NixOS)
- Version: [1.7.1](https://github.com/NixOS/nixpkgs/blob/nixos-19.09/pkgs/tools/text/vale/default.nix#L5)

When I reference a style that does not have a corresponding directory, `vale` ignores this silently.

Example `.vale.ini`:

```ini
StylesPath = .local/share/vale-styles
MinAlertLevel = suggestion
 
[*.{md}]
BasedOnStyles = style-that-does-not-exist
```

```console
$ vale .
✔ 0 errors, 0 warnings and 0 suggestions in 74 files.
```

I expect `vale` to error out in this case. Suppressing this kind of error makes it more difficult to configure `vale` correctly.

**Comment by jdkato** [comment]
This has been addressed by v2.0's `--debug` flag.

**Comment by curiousleo** [comment]
I struggle to think of a situation where I would _not_ want to be informed that my configuration is invalid, so it is surprising to me that I need to pass `--debug` to be told that my configuration doesn't work.

I appreciate the quick response though.

**Comment by curiousleo** [comment]
> This has been addressed by v2.0's `--debug` flag.

I can't replicate that. I've built `vale` from `master` (9ce917fd6f68849a2a6eb5338f978709e8ec3096), and I see no change. I'm using the same `ini` file as in the ticket description.

The ini file is read and parsed correctly, but shows no error that `style-that-does-not-exist` does not exist. There is no `--debug` flag for the `dc` command.

```console
$ ~/Code/vale/bin/vale dc
{
  "Checks": null,
  "GBaseStyles": [],
  "GChecks": {},
  "IgnoredScopes": null,
  "SkippedScopes": null,
  "BlockIgnores": {},
  "TokenIgnores": {},
  "MinAlertLevel": 0,
  "RuleToLevel": {},
  "SBaseStyles": {
    "*.{md}": [
      "style-that-does-not-exist"
    ]
  },
  "SChecks": {
    "*.{md}": {}
  },
  "StylesPath": "/home/leo/.local/share/vale-styles",
  "WordTemplate": "",
  "Parsers": {},
  "Formats": {},
  "Path": "/home/leo",
  "SecToPat": {
    "*.{md}": {
      "Value": {
        "Matchers": [
          {
            "Str": ".",
            "RunesLength": 1,
            "BytesLength": 1,
            "Segments": [
              1
            ]
          },
          {
            "Str": "md",
            "RunesLength": 2,
            "BytesLength": 2,
            "Segments": [
              2
            ]
          }
        ],
        "RunesLength": 3,
        "Segments": [
          3
        ]
      },
      "Left": {},
      "Right": null,
      "ValueLengthRunes": 3,
      "LeftLengthRunes": -1,
      "RightLengthRunes": 0,
      "LengthRunes": -1
    }
  },
  "Output": "CLI",
  "Wrap": false,
  "NoExit": false,
  "Sorted": false,
  "Normalize": false,
  "Simple": false,
  "InExt": ".txt",
  "Relative": false
}
```

When I apply `vale` to a file, there is no error even with `--debug`:

```console
$ ~/Code/vale/bin/vale test.md --debug
✔ 0 errors, 0 warnings and 0 suggestions in 1 file.
```

As I understood your comment @jdkato, I should now see an error message, but it's not there.

**Comment by jdkato** [comment]
`master` is stll v1.7.1; you can find the compiled binaries for v2.0 on the [releases page](https://github.com/errata-ai/vale/releases/tag/v2.0.0-beta.2).

You should then see something along the lines of the following:

```console
$ vale --debug test.md
lstat build/styles/style-that-does-not-exist: no such file or directory
✔ 0 errors, 0 warnings and 0 suggestions in 1 file.
```
(Note that `--debug` is passed *prior* to the file argument.)

> I struggle to think of a situation where I would not want to be informed that my configuration is invalid, so it is surprising to me that I need to pass --debug to be told that my configuration doesn't work.

I don't necessarily disagree, but I also want to ensure that `--output` is always respected by default (i.e., I don't want return invalid JSON when errors/warnings are reported). This involves a code refactor that I haven't had time to do yet.

**Comment by curiousleo** [comment]
Awesome, thanks for the help @jdkato.

I've been able to track down the issue I'm having. Since `vale --debug <file>` (on v2.0.0-beta.2) was showing neither errors nor lints for a file I had specifically created to be full of lint errors, I used `strace`.

Here's a relevant line:
```
newfstatat(AT_FDCWD, "/home/leo/.local/share/vale-styles/Microsoft", {st_mode=S_IFLNK|0777, st_size=25, ...}, AT_SYMLINK_NOFOLLOW) = 0
```

`AT_SYMLINK_NOFOLLOW` looks very much like it is the issue: because the lints I want are in my dotfiles repo, which is integrated into my system using symlinks, `vale` fails to read them:

```console
$ ls -l ~/.local/share/vale-styles
lrwxrwxrwx 1 leo users 49 Nov  6 20:09 /home/leo/.local/share/vale-styles -> ../../.setup/config/vale/.local/share/vale-styles
```

At least I understand what the problem is now. I may follow this up with a FR to resolve symlinks for styles directories.

Thanks again for your help!
